x-pack/test_serverless/api_integration/test_suites/observability/custom_threshold_rule/avg_pct_no_data.ts

```diff
@@ -51,10 +48,6 @@ export default function ({ getService }: FtrProviderContext) {
         index: CUSTOM_THRESHOLD_RULE_ALERT_INDEX,
         query: { term: { 'kibana.alert.rule.uuid': ruleId } },
       });
-      await esClient.deleteByQuery({
```

Comment:
Any particular reason to remove cleaning up the event log index?

Reply:
But, the event log index is related to the result of executing a rule, not the ingested data.

You can check the test server with and without cleaning; you'll see without it, you still have some logs in this index.

![image](https://github.com/elastic/kibana/assets/12370520/6e80e078-3b26-4301-afcb-e6a04e0c5802)

Reply:
@maryam-saeidi thank you for pointing that out. There is a race condition on this index as we use the wildcard option `*`, which creates inconsistent results and sometimes throws an error. 

`"cause":{"type":"version_conflict_engine_exception","reason":"[S_D574oB8qizAIQCLOib]: version conflict, required seqNo [1], primary term [1]. but no document was found","index_uuid":"cZ1A-w-KS1qY34KrbPGHPA","shard":"0","index":".ds-.kibana-event-log-ds-2023.10.02-000001"},"status":409},{"index":".ds-.kibana-event-log-ds-2023.10.02-000001","id":"TPD574oB8qizAIQCMOiB","cause":{"type":"version_conflict_engine_exception","reason":"[TPD574oB8qizAIQCMOiB]: version conflict, required seqNo [2], primary term [1]. but no document was found","index_uuid":"cZ1A-w-KS1qY34KrbPGHPA","shard":"0","index":".ds-.kibana-event-log-ds-2023.10.02-000001"},"status":409}`

So, I changed the delete term to use the `rule.id` instead of the `alert.consumer` to be more accurate and avoid this issue.

Reply:
@maryam-saeidi No, the error message is before I change it to `rule.id`

> The error says there is no data based on the query,

This is the way I removed the `deleteByQuery`, but then I realized there are a racing condition.
If you look at the error type `version_conflict_engine_exception`. It means there are two operations at the same.

Reply:
> This is why I removed the deleteByQuery in the beginning, but then I realized there is a racing condition.
If you look at the error type version_conflict_engine_exception. It means there are two operations at the same.

I still don't understand, either the document exists in the index, which should contain both fields (`rule.id` and `kibana.alert.rule.consumer`), or the document does not exist. How does checking one field instead of the other help?
In which case, do we have a document with `rule.id` that does not contain `kibana.alert.rule.consumer`?

Reply:
Yes, super clear, thanks.

Then can we bring back cleaning the event log index with the `rule.id` condition? Otherwise, after this test, we still have data in the event log index.

```
await esClient.deleteByQuery({
  index: '.kibana-event-log-*',
  query: { term: { 'rule.id': ruleId } },
});
```

Reply:
Yeah, I already got it back!